extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQSelectTest.java

```diff
@@ -1467,22 +1467,21 @@ public void testExternSelectWithMultipleWorkers() throws IOException
         );
     // adding result stage counter checks
     if (isPageSizeLimited()) {
-      selectTester = selectTester.setExpectedCountersForStageWorkerChannel(
+      selectTester.setExpectedCountersForStageWorkerChannel(
           CounterSnapshotMatcher
-              .with().rows(2, 0, 2).frames(1, 0, 1),
+              .with().rows(2, 0, 2),
```

Comment:
Let's not remove the number of frames assertion for the UTs to pass, since that shouldn't be flaky.

If they produce different results with sql-compat and sql-incompat mode, we should have separate assertions based on the mode it is run.

Reply:
The issue is with how frameProccessors are getting scheduled and not the SQL compat mode. 
Since all other PR's are failing, ie a dev blocker, lets merge it till I find a better way to solve it?

Reply:
Sounds good